src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -1028,7 +1028,7 @@ private static string InferTupleElementName(SyntaxNode syntax)
             string name = syntax.TryGetInferredMemberName();
 
             // Reserved names are never candidates to be inferred names, at any position
-            if (name == null || TupleTypeSymbol.IsElementNameReserved(name) != -1)
+            if (name == null || NamedTypeSymbol.IsElementNameReserved(name) != -1)
```

Comment:
>IsElementNameReserved [](start = 48, length = 21)

Suggest rename to `IsTupleElementNameReserved`.  You might consider moving these tuple-specific methods and factories to a static class `TupleTypeSymbol` for clarity.